numba/tests/test_funcdesc.py

```diff
@@ -0,0 +1,29 @@
+
+import unittest
+from numba import njit
+
+
+class TestModule(unittest.TestCase):
+    def test_module_not_in_namespace(self):
+        """ Test of trying to run a compiled function
+        where the module which the function is being compiled
+        doesn't exist in the namespace.
+        """
+        filename = 'test.py'
+        name = 'mypackage'
+        code = """
+def f(x):
+    return x
+"""
+
+        objs = dict(__file__=filename, __name__=name)
+        compiled = compile(code, filename, 'exec')
+        exec(compiled, objs)
+
+        compiled_f = njit(objs['f'])
+        with self.assertRaises(ModuleNotFoundError):
+            compiled_f(0)
```

Comment:
This pattern is used in Numba tests to ensure that the raised exception is the one expected. i.e. `with self.assertRaises(<exception class>) as raises: ... self.assertIn(<some message>, str(raises.exception))`
https://github.com/numba/numba/blob/82c7d37ad4900b01072c233da5ea97e95d5444db/numba/tests/test_array_analysis.py#L455-L460

Reply:
updated in dc22351

Reply:
Great, thank you for doing that.